Build fails when not using a git checkout

### Description
When downloading a zip or tarball of the source (eg https://github.com/PegaSysEng/teku/archive/0.11.3.zip) the build fails with:

```
FAILURE: Build failed with an exception.

* Where:
Build file '/Users/aj/Downloads/teku-0.11.3/build.gradle' line: 588

* What went wrong:
A problem occurred evaluating root project 'teku'.
> Cannot invoke method head() on null object

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 9s
```

This is because the git version was not able to be calculated.  We need to detect this situation and fallback to using a string like "&lt;UNKNOWN>".

--- a/build.gradle
+++ b/build.gradle
@@ -585,7 +585,7 @@ def calculateVersion() {
 
 def getCheckedOutGitCommitHash() {
   def takeFromHash = 8
-  grgit.head().id.take(takeFromHash)
+  grgit ? grgit.head().id.take(takeFromHash) : 'UNKNOWN'
 }
 
 task jacocoRootReport(type: JacocoReport) {

